tests/integration/src/test/java/org/apache/pulsar/tests/integration/cli/topicpolicies/SchemaCompatibilityStrategyTest.java

```diff
@@ -19,67 +19,68 @@
 
 package org.apache.pulsar.tests.integration.cli.topicpolicies;
 
-import static org.testng.Assert.assertTrue;
+import static org.testng.Assert.assertEquals;
 import org.apache.pulsar.common.policies.data.SchemaCompatibilityStrategy;
 import org.apache.pulsar.tests.integration.docker.ContainerExecResult;
-import org.apache.pulsar.tests.integration.suites.PulsarTestSuite;
+import org.apache.pulsar.tests.integration.suites.PulsarCliTestSuite;
 import org.awaitility.Awaitility;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
 import org.testng.annotations.Test;
 
-public class SchemaCompatibilityStrategyTest extends PulsarTestSuite {
+public class SchemaCompatibilityStrategyTest extends PulsarCliTestSuite {
+    @BeforeMethod(alwaysRun = true)
```

Comment:
Is this method supposed to be annotated with `@BeforeClass`? `super.before()` is annotated that way. Given that `enableTopicPolicies()` sets broker config and brokers are only set up once per class, I think we want `@BeforeClass`. 

```suggestion
    @BeforeClass(alwaysRun = true)
```

Reply:
There are some tests that will affect the results of another test, like I set schema compatibility strategy at namespace level in `testSchemaCompatibilityStrategyCmdWithNamespaceLevel`,  if this test failed, it will break the `testSchemaCompatibilityStrategyCmd` test result, so I use `@BeforeMethod(alwaysRun = true)` instead of `@BeforeClass(alwaysRun = true)` to avoid this thing, each test have own Pulsar is safe.

Reply:
Okay, I can see how state of individual tests can impact each other. However, the current design of the `PulsarCliTestSuite` does not align with its usage here. Is there a way to ensure test isolation without resetting the cluster? If not, I think it'd be better to make it clearer that the `PulsarCliTestSuite#before` method should be called before each test and that the abstract suite is designed to make individual tests isolated.

A secondary concern is that it has to be somewhat expensive to reset brokers. We should try to limit the cost of basic tests, if possible.

Reply:
Your worry is necessary. I know that reset broker is somewhat expensive. The `PulsarCliTestSuite` is abstract for cli test, normal we don't override the `before` and `after` methods, which need to be overridden when tests need isolation.

Reply:
@michaeljmarshall Can we unblock the 2.10.0 release first and create a new issue to track the test improvement?

Reply:
Sure, let's create an issue.